docs/sources/clients/promtail/logrotation.md

```diff
@@ -0,0 +1,137 @@
+# Promtail and Log Rotation
+
+## Why does log rotation matters?
+
+At any point in time, there may be three processes working on a log file as shown in the image below.
+![block_diagram](./logrotation-components.png)
+
+1. Appender - A writer that keeps appending to a log file. This can be your application or some system daemons like Syslog, Docker log driver or Kubelet, etc.
+2. Tailer - A reader that reads log lines as they are appended e.g agents like Promtail.
+3. Log Rotater - A process that rotates the log file either based on time (say scheduled every day) or size (say a log file reached its max size).
+
+> **NOTE:** Here `fd` defines a file descriptor. Once a file is open for read or write, OS gives back a unique file descriptor (usually an integer) per process, and all the operations like read and write are done over that file descriptor. In other words, once the file is opened successfully, the file descriptor matters more than the file name.
+
+One of the critical components here is the log rotater. Let's understand how it impacts other components like the appender and tailer.
+
+In general, when a log is rotated, it can be done in two ways.
```

Comment:
This is passive voice, could we make it active to indicate who or what is doing the rotating?  Is this an action that users have to take, or does it happen automatically?

In general, when a software program rates a log, it can be done in two ways.
In general, when Promtail rotates a log, it can be done in two ways.

Reply:
> This is passive voice, could we make it active to indicate who or what is doing the rotating? Is this an action that users have to take, or does it happen automatically?

It happens automatically (internal to log rotator).

I like "In general, when a software program rotates a log, it can be done in two ways internally."

Reply:
> I like "In general, when a software program rotates a log, it can be done in two ways internally."

Then I'd use that wording.

Reply:
Nice. updated already :+1: